bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/JournalChannel.java

```diff
@@ -299,6 +298,8 @@ int read(ByteBuffer dst)
     public void close() throws IOException {
         if (bc != null) {
             bc.close();
+        } else if (fc != null) {
```

Comment:
I would remove "else" and handle the 2 variables indipendenly

also we should try to close both of them in spite of any errors


```
try {
 if (bc != null) {
     bc.close();
 } 
} finally {
     if (fc != null) {
         fc.close();
     }     
} 

```

Reply:
@eolivelli bc.close will close the ubder FileChannel, double close may raise some other exception